pkg/expr/evaluator.go

```diff
@@ -42,9 +42,6 @@ type (
 
 	// Validator validates a given expression
 	Validator interface {
```

Comment:
if we are removing `Validate()` from `Validator`, should we also change the name of the interface to something like `TypeChecker` (and maybe update the method `TypeCheck` to `CheckType`)?